projects/core/src/i18n/i18next/i18next-plugins/i18next-logger-plugin.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { InjectionToken, inject } from '@angular/core';
+
+import { LoggerModule } from 'i18next';
+import { LoggerService } from '../../../logger';
+
+//TODO: add unit tests if approved
+export const I18NEXT_LOGGER_PLUGIN = new InjectionToken<LoggerModule>(
+  'I18NEXT_LOGGER_PLUGIN',
+  {
+    providedIn: 'root',
+    factory: () => {
+      const logger = inject(LoggerService);
+      return {
+        type: 'logger',
+        log: (args) => logger.log(...args),
```

Comment:
Regarding https://github.com/SAP/spartacus/pull/17441#discussion_r1219171868:

when I used spread operator and then simply log to the console, as you suggested, 
```
    return {
        type: 'logger',
        log: (...args) => {
          console.log(...args);
          logger.log(...args);
        },
        warn: (args) => logger.warn(...args),
        error: (args) => logger.error(...args),
    };
```
it logged an array... 
![image](https://github.com/SAP/spartacus/assets/28891782/cbd20f0a-eae5-4c4a-9bb0-f0bae9cda70d)

maybe it's about how i18next is passing args to the plugin? Anyway spread operator on the function's args is messing up the output.

Reply:
It looks like i18next is always passing one character which is an array (that can contain many values) - [source code](https://github.com/i18next/i18next/blob/5de7d4687bdf03dd5c5b4938d8bfb8db78dc4720/i18next.js#L34-L35)
Because of that, if we use the spread operator on args in the arrow function, we're saying that there may be more than one argument, that's why logger (or console.log) outputs an array when we use to spread on provided args again. See in example:
![image](https://github.com/SAP/spartacus/assets/28891782/89a0f51e-a5cd-465d-83e1-298576603eeb)

Reply:
As we discussed offline, i18next expects just a single agrument `args` which is an array of parameters to be passed to a concrete logger (which was suprising for us).